packages/react-utilities/src/utils/properties.ts

```diff
@@ -380,59 +365,47 @@ export const imgProperties = toObjectMap(htmlElementProperties, [
 ]);
 
 /**
- * @deprecated Use imgProperties for img elements.
- */
-export const imageProperties = imgProperties;
-
-/**
- * An array of DIV tag properties and events.
- *
- * @public
+ * A set of DIV tag properties and events.
  */
 export const divProperties = htmlElementProperties;
 
 /**
- * Gets native supported props for an html element provided the allowance set. Use one of the property
- * sets defined (divProperties, buttonPropertes, etc) to filter out supported properties from a given
- * props set. Note that all data- and aria- prefixed attributes will be allowed.
- * NOTE: getNativeProps should always be applied first when adding props to a react component. The
- * non-native props should be applied second. This will prevent getNativeProps from overriding your custom props.
- * For example, if props passed to getNativeProps has an onClick function and getNativeProps is added to
- * the component after an onClick function is added, then the getNativeProps onClick will override it.
+ * Gets native supported props for an HTML element provided the allowance set. Use one of the property
+ * sets defined (`divProperties`, `buttonPropertes`, etc) to filter out supported properties from a given
+ * props set. Note that all `data-` and `aria-` prefixed attributes will be allowed.
  *
- * @public
+ * NOTE: `getNativeProps` should always be applied first when adding props to a React component. The
+ * non-native props should be applied second. This will prevent `getNativeProps` from overriding your custom props.
+ * For example, if props passed to `getNativeProps` has an `onClick` function and `getNativeProps` is added to
+ * the component after an `onClick` function is added, then the `getNativeProps` `onClick` will override it.
  * @param props - The unfiltered input props
- * @param allowedPropsNames - The array or record of allowed prop names.
+ * @param allowedPropsNames - The set of allowed prop names.
  * @returns The filtered props
  */
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 export function getNativeProps<T extends Record<string, any>>(
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   props: Record<string, any>,
-  allowedPropNames: string[] | Record<string, number>,
+  allowedPropNames: Set<string>,
   excludedPropNames?: string[],
+  extraPropNames?: string[],
```

Comment:
we'll probably always need to make an exception for the `as` prop, we could make it a default param

```suggestion
  extraPropNames: string[] = ['as'],
```

Reply:
Adding a default param `= ['as']` could potentially cause unexpected bugs. E.g. say I have this code:
`getNativeProps(props, allowedPropNames);`
By default, it is adding `'as'` as an extra prop.

But now I want to also add `'foo'` as an extra prop, so I add the extraPropNames param:
`getNativeProps(props, allowedPropNames, [], ['foo']);`
💣 Now it stops including `'as'` because it overwrites the default param.